ironfish-cli/src/utils/transaction.ts

```diff
@@ -36,11 +36,43 @@ export async function watchTransaction(options: {
 
   const startTime = lastTime
 
-  if (last.content.transaction == null) {
-    logger.log(`Tried to watch transaction ${options.hash} but it's missing.`)
-    return
+  let state = ''
```

Comment:
There are now 2 states, when there used to be 1 state. Do we need 2 states?

Reply:
I'll refactor the old code to use the new state system. I needed to introduce new state to know if the while loop was entered